Raise exception on permission-related rejection

dvejmz opened the pull request:
Closes #203 

Raise permission exception if a publish via the
Twisted API was rejected due to insufficient
permissions

Signed-off-by: dvejmz <dvejmz@gmail.com>

jeremycline reviewed (dismissed):
Generally this looks great, thanks for the pull request!

I'd really like to see an integration test to go along with it, there are a bunch of tests for the API in https://github.com/fedora-infra/fedora-messaging/blob/master/fedora_messaging/tests/integration/test_api.py.

dvejmz commented:
@jeremycline sure, I'll write one, leave it with me!

abompard commented:
Hey @dvejmz , are you still interested in writing this integration test?

dvejmz commented:
Hi @abompard , I do apologise, this one definitely slipped through the cracks, I'd completely forgotten about this! I'll rebase this off master and write the integration test this week.

abompard commented:
Hey! No problem. I recently fixed an issue that kinda looks similar but not entirely, so you can reuse the `PublishForbidden` exception I've added.

dvejmz commented:
Hey guys @jeremycline @abompard , I've (finally!) now rebased this branch off the latest `stable` while making some changes to the original code I'd written:

* Changed the exception handler to raise the new `PublishForbidden` exception added by @abompard rather than the exception I had created
* Changed the `publish()` unit test I'd originally written to be more relevant
* Added the requested API integration test

Let me know if you have any other comments.

abompard reviewed (changes requested):
Looks great! Could you please add a news fragment for the release notes? It's described [here](https://fedora-messaging.readthedocs.io/en/stable/contributing.html#release-notes). Thanks!

dvejmz commented:
All done @abompard!

abompard commented:
Thanks, but the `news/203.feature` file is empty. It should contain the snippet you want to appear in the release notes, for example "Raise PublishForbidden exception if publishing to channel is denied".

dvejmz commented:
Ah, I see, I'm not familiar with the process, sorry about that. I'll get this fixed later today.

dvejmz commented:
I've now updated the news fragment with the requested snippet. Thanks for your comments and patience! @abompard . I'll be sure to remember this next time.

abompard commented on news/203.feature:
```diff
@@ -0,0 +1 @@
+Raise `PublishForbidden` exception immediately if publishing to [virtual host](https://www.rabbitmq.com/vhosts.html) is denied rather than waiting until timeout.
```
Thanks! Unfortunately this will be inserted in a RestructuredText document so this has to RST and not Markdown... sorry!

dvejmz replied:
Fixed

abompard reviewed (approved):
Looks great, thanks a lot David!
